ts/store/reducers/entities/messages/messagesStatus.ts

```diff
@@ -163,4 +163,19 @@ export const isMessageRead = (
     .map(ms => ms.isRead)
     .getOrElse(false);
 
+/**
+ * Retrieve the MessageStatus given an ID.
+ * Fallback to all false if not found.
+ *
+ * @param state
+ * @param messageId
+ */
+export const getMessageStatus =
+  (state: GlobalState) =>
+  (messageId: string): MessageStatus =>
+    fromNullable(state.entities.messagesStatus[messageId]).getOrElse({
+      isRead: false,
+      isArchived: false
+    });
```

Comment:
```suggestion
export const getMessageStatus = createSelector(
  [messagesStatusSelector, (_: GlobalState, messageId: string) => messageId],
  (messagesStatus, messageId) =>
    fromNullable(messagesStatus[messageId]).getOrElse({
      isRead: false,
      isArchived: false
    })
);
```
you could use `createSelector` to take advantage of memoization
https://github.com/reduxjs/reselect#q-how-do-i-create-a-selector-that-takes-an-argument
If you like this suggestion you have to change the usage in `mapStateToProps`
from
`getMessageStatus: getMessageStatus(state),`
to
`getMessageStatus: (messageId: string) => getMessageStatus(state, messageId),`

Reply:
Good point! Added [here](https://github.com/pagopa/io-app/pull/3526/files#diff-f10f8d26976cd0b78e2afdfa68b2ece7561c6337b55331e2d8b12953807350cbR173)